jck/update_git_material.sh

```diff
@@ -0,0 +1,145 @@
+perform_hard_reset_and_pull() {
+    echo "Performing hard reset..."
+    git reset --hard "origin/$jck_branch" || {
+        echo "Hard reset failed. Removing directory and cloning again."
+        remove_and_clone
+    }
+
+    # Clean untracked files and directories
+    git clean -dfx || {
+        echo "Failed to clean untracked files and directories."
+        remove_and_clone
+    }
+
+    # Pull the latest changes
+    git pull "$JCK_GIT_REPO_USED" "$jck_branch" || {
+        echo "Pull failed. Removing directory and cloning again."
+        remove_and_clone
+    }
+}
+
+# Function to check for updates
+check_and_handle_updates() {
+    local localSHA=$(git rev-parse HEAD)
+    localSHATrimmed=$(echo "${localSHA}" | tr -d '[:space:]')
+    local remoteSHA=$(git ls-remote "$JCK_GIT_REPO_USED" "$jck_branch" | cut -f1)
+
+    echo "LocalSHA = --$localSHA--"
+    echo "RemoteSHA = --$remoteSHA--"
+
+    # Check if SHA match
+    if [ "$localSHA" != "$remoteSHA" ]; then
+        # Update materials
+        delete_native_dirs
+        echo message="Updating ${JCK_ROOT_USED} with latest..."
+        cd "${JCK_ROOT_USED}" || exit
+        git pull "$JCK_GIT_REPO_USED" "$jck_branch" || {
+            perform_hard_reset_and_pull
+        }         
+    else
+        # Set the property indicating that local materials are up-to-date
+         echo "Local JCK materials up-to-date. Skipping update"
+        local_material_uptodate="true"
+    fi
+
+    # If SHA match or update succeeded
+   echo "Local JCK materials updated"
+
+    # Check if isZOS property is set
+    if [ -n "$isZOS" ]; then
+        # Check if local-material-uptodate property is set
+        if [ -n "$local_material_uptodate" ]; then
+            echo "zOS Local JCK materials up-to-date. Skipping hard reset"
+        else
+            # Perform hard reset for z/OS
+            perform_hard_reset_zos
+        fi
+    fi
+}
+
+# Function to perform hard reset for z/OS
+perform_hard_reset_zos() {
+    echo "Performing hard reset of ${JCK_ROOT_USED} using .gitattributes.zos for file conversions..."
+    rm -rf "${JCK_ROOT_USED}/.git/info"
+    mkdir -p "${JCK_ROOT_USED}/.git/info"
+    mv "${JCK_ROOT_USED}/.gitattributes.zos" "${JCK_ROOT_USED}/.git/info/attributes"
+    git rm --cached -r -q .
+    git reset --hard
+}
+
+# Check if JCK materials exist
+if [ -d "$JCK_ROOT_USED" ]; then
+    # Change into the directory
+    cd "$JCK_ROOT_USED" || {
+        echo "Failed to change directory to $JCK_ROOT_USED"
+        exit 1
+    }
+    # Check if it's a valid Git repository
+    if [ -d "$JCK_ROOT_USED/.git" ] || git -C "$JCK_ROOT_USED" rev-parse --is-inside-work-tree &>/dev/null; then
+        echo "The directory is a valid Git repository."
+        check_and_handle_updates
+    else
+        echo "The directory is not a valid Git repository."
+        remove_and_clone
+    fi    
+else
+    # JCK materials don't exist, git clone
+    git_clone || remove_and_clone
```

Comment:
If the directory doesn't exist, `git_clone` once would be enough, no need `remove_and_clone`.